Vineyard/Vineyard/Badge.swift

```diff
@@ -6,22 +6,42 @@
 //
 
 import Foundation
+import FirebaseFirestore
 
 struct Badge: Identifiable, Codable {
-    let id: String = UUID().uuidString
-    var resolution: Resolution
-    var group: Group
+    @DocumentID var id: String? = UUID().uuidString
```

Comment:
I don't think we need an initial value for anything marked with `@DocumentID`. I'm seeing this warning in the console:
```
The document ID is managed by Firestore and any initialized or set value will be ignored. 
The ID is automatically set when reading from Firestore.
```

```suggestion
    @DocumentID var id: String?
```